azpubsub/contextvalidator/src/test/java/AzPubSubPrincipalBuilderTest.java

```diff
@@ -0,0 +1,161 @@
+        InetAddress inetAddress =  InetAddress.getLoopbackAddress();
+
+        azPubSubPrincipalBuilder.configure(configs);
+
+        SslAuthenticationContext sslAuthenticationContext = new SslAuthenticationContext(sslSession, inetAddress);
+        KafkaPrincipal kafkaPrincipal = azPubSubPrincipalBuilder.build(sslAuthenticationContext);
+        assert(kafkaPrincipal.getName().equals("ANONYMOUS"));
+        assert(KafkaPrincipal.ANONYMOUS.getPrincipalType().equals(kafkaPrincipal.getPrincipalType()));
+        assert(KafkaPrincipal.ANONYMOUS.getName().equals(kafkaPrincipal.getName()));
+    }
+
+    @Test
+    public void testValidateSaslAuthenticationContextPositive() {
+        Map<String, Object> configs = new java.util.HashMap<>();
+
+        configs.put("azpubsub.ssl.authentication.validator.class", "mockPositiveSslAuthenticationContextValidator");
+        configs.put("azpubsub.sasl.authentication.validator.class", "mockPositiveSaslAuthenticationContextValidator");
+        MemberModifier.suppress(MemberMatcher.methodsDeclaredIn(Logger.class));
+
+        SaslServer saslServer = EasyMock.mock(SaslServer.class);
+        InetAddress inetAddress =  InetAddress.getLoopbackAddress();
+        azPubSubPrincipalBuilder.configure(configs);
+
+        SaslAuthenticationContext saslAuthenticationContext = new SaslAuthenticationContext(saslServer, SecurityProtocol.SASL_SSL, inetAddress);
+        KafkaPrincipal kafkaPrincipal = azPubSubPrincipalBuilder.build(saslAuthenticationContext);
+        assert(kafkaPrincipal.getPrincipalType().equals("Token"));
+    }
+
+    @Test
+    public void testValidateSaslAuthenticationContextInvalidAuthentication() {
+        expectedEx.expect(IllegalSaslStateException.class);
+        expectedEx.expectMessage("failed to authenticate the current context");
+
+        Map<String, Object> configs = new java.util.HashMap<>();
+
+        configs.put("azpubsub.ssl.authentication.validator.class", "mockPositiveSslAuthenticationContextValidator");
+        configs.put("azpubsub.sasl.authentication.validator.class", "mockSaslAuthenticationContextValidatorWithInvalidAuthentication");
+        MemberModifier.suppress(MemberMatcher.methodsDeclaredIn(Logger.class));
+
+        SaslServer saslServer = EasyMock.mock(SaslServer.class);
+        InetAddress inetAddress =  InetAddress.getLoopbackAddress();
+        azPubSubPrincipalBuilder.configure(configs);
+
+        SaslAuthenticationContext saslAuthenticationContext = new SaslAuthenticationContext(saslServer, SecurityProtocol.SASL_SSL, inetAddress);
+        KafkaPrincipal kafkaPrincipal = azPubSubPrincipalBuilder.build(saslAuthenticationContext);
+    }
+
+    @Test
+    public void testValidateSslAuthenticationContextWithInvalidAuthentication() {
+        expectedEx.expect(IllegalStateException.class);
+        expectedEx.expectMessage("Ssl Authentication Context Validator failed to validate the current SSL session");
```

Comment:
The expected message is problematic. Where does it come from? It should be constant.

Reply:
This message from dSTS Authentication library (C#). What you see is a substring. I used to encounter an issue that the upgrade of the library changed the error message a bit by changing the error message, but this part is something common and consistent.